### ISSUE TITLE
a11y: Column and row counts on the `KeyValue` are incorrect when they are interpreted with a screen reader

### ISSUE THREAD

**Issue by aalves08** [rancher/dashboard#14439]
**Setup**
- Rancher version: head, any
- Rancher UI Extensions:
- Browser type & version:

**Describe the bug**

Column and row counts on the `KeyValue` are incorrect when they are interpreted with a screen reader

We need to address this to ensure that the column counts are accurate. Right now, we have situations where the column is called out as "Column3" but there is no column 1 or 2.

**To Reproduce**

Use a screen reader on a `KeyValue` component, check output

**Result**
Right now, we have situations where the column is called out as "Column3" but there is no column 1 or 2.

**Expected Result**

Column and row counts should be correct when using a screen reader

**Screenshots**

**Additional context**

**Comment by rak-phillip** [comment]
I think this needs some more triage to gather some more information about the design of the key/value component and the accessibility implementation.

**Comment by nwmac** [comment]
Do we plan to do this?

**Comment by rak-phillip** [comment]
I think that it would be good to at least do the proper triage and narrow down the acceptance criteria for this issue.
